repository/src/main/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelper.java

```diff
@@ -371,28 +371,20 @@ public List<RepositoryFile> getAllDeletedFiles( final Session session,
     if ( isAdmin() ) {
       List<RepositoryFile> deletedFiles = new ArrayList<>();
       ITenant tenant = JcrTenantUtils.getTenant();
-      // BISERVER-14482: this user list refers to Jcr, as so, if authentication/authorization is otherwise managed
+      // (deprecated comment) BISERVER-14482: this user list refers to Jcr, as so, if authentication/authorization is otherwise managed
       // (LDAP, for instance), one has to guarantee that the current user is on this list!
       // The correct way would be to fix the list by obtaining the LDAP users, but this has the potential of getting
       // thousands of users and, consequently, thousands of deleted files!
```

Comment:
> The correct way would be to fix the list by obtaining the LDAP users, but this has the potential of getting thousands of users and, consequently, thousands of deleted files!

Can't we incur into this issue with the new code? Seems like other users files not appearing was the expected behaviour due to this reason. Perhaps there should be a filter/flag to toggle visibility of other users deleted files?

Reply:
This issue states that this is not the correct behavior. In the case of LDAP, the admin should be able to clean the trash folder.

Reply:
Not having a user able to delete the trash is causing clients with LDAP to have infinitely increasing repositories without being able to clean it (several tickets identified)

Reply:
Just raising the question because I wondered if there was a reason of why that comment was added that would break PUC for the admin user in that scenario

Reply:
The code change looks 👌

Reply:
The problem here was more on the "potential of getting thousands of users": thinking of a large organization, the list would be huge while the number of **actual** Pentaho users would be small.
Using the previous strategy of inquiring if a user existed and if it had trash, would mean that most of these inquiries to be useless, but yet, consume much time...
As for the "consequently, thousands of deleted files", well... if that's the reality, it has to be shown. We'll have to wait for concrete problems in this area to see how we can mitigate the scenario Bruno is describing happens (if it does happen).